core-services/src/main/java/org/zalando/nakadi/service/SchemaService.java

```diff
@@ -232,14 +233,23 @@ private void validateJsonTypeSchema(final EventTypeBase eventType, final String
             throw new SchemaValidationException("\"metadata\" property is reserved");
         }
 
+        final JSONObject effectiveSchemaAsJson = jsonSchemaEnrichment.effectiveSchema(eventType, eventTypeSchema);
+        final Schema effectiveSchema = SchemaLoader.load(effectiveSchemaAsJson);
+
+        final EventOwnerSelector eventOwnerSelector = eventType.getEventOwnerSelector();
+        if (eventOwnerSelector != null) {
+            if (eventOwnerSelector.getType() == EventOwnerSelector.Type.PATH) {
+                validateFieldsInSchema("event_owner_selector.value", List.of(eventOwnerSelector.getValue()),
```

Comment:
Since we are not blocking publishing for existing event type, does it also make sense to not block updating for existing event types? maybe we could introduce this for newly created event types and/or event types which are already valid should not become invalid.

Reply:
In general the migration can be more gradual, but here we know that only 4 event types need to be updated — that should be pretty straightforward.